#! /bin/sh -e

# DP: Misc. tidbits (mostly compiler warnings)

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- Makeconfig~ 	2001/09/12 18:49:45	1.265
+++ Makeconfig 	2001/10/07 20:56:17
@@ -532,9 +532,10 @@
 $(subst $(empty) ,:,$(strip $(patsubst -Wl$(comma)-rpath-link=%, %,\
 				       $(filter -Wl$(comma)-rpath-link=%,\
 						$(sysdep-LDFLAGS)))))
+ld_envlib_path = $(shell test x$$LD_LIBRARY_PATH = x || echo -n :$$LD_LIBRARY_PATH)
 run-program-prefix = $(if $(findstring $(notdir $(built-program-file)), $(tests-static)),, \
 			  $(elf-objpfx)$(rtld-installed-name) \
-			  --library-path $(rpath-link)$(patsubst %,:%,$(sysdep-library-path)))
+			  --library-path $(rpath-link)$(patsubst %,:%,$(sysdep-library-path)))$(ld_envlib_path)
 else
 run-program-prefix =
 endif
--- localedata/locales/es_AR~ 	2000/10/26 23:22:55	1.10
+++ localedata/locales/es_AR 	2001/10/07 20:56:18
@@ -80,8 +80,8 @@
 
 LC_NUMERIC
 decimal_point        "<U002C>"
-thousands_sep        ""
-grouping             0;0
+thousands_sep        "<U002E>"
+grouping             3;3
 END LC_NUMERIC
 
 LC_TIME
--- sysdeps/i386/fpu/bits/mathinline.h~ 	2001/07/06 04:55:53	1.48
+++ sysdeps/i386/fpu/bits/mathinline.h 	2001/10/07 20:56:21
@@ -381,7 +381,7 @@
     ("fscale			# 2^int(x * log2(e))\n\t"		      \
      : "=t" (__temp) : "0" (1.0), "u" (__exponent));			      \
   __temp -= 1.0;							      \
-  return __temp + __value ?: __x
+  return __temp + __value ? __temp + __value : __x
 __inline_mathcodeNP_ (long double, __expm1l, __x, __expm1_code)
 
 
--- elf/ldconfig.c~	2001/08/23 23:31:17
+++ elf/ldconfig.c	2001/10/27 14:03:48
@@ -906,26 +906,24 @@
 {
   FILE *file = NULL;
   char *line = NULL;
-  const char *canon;
+  const char *canon = filename;
   size_t len = 0;
+  int file_fd;
 
   if (opt_chroot)
     {
       canon = chroot_canon (opt_chroot, filename);
-      if (canon)
-	file = fopen (canon, "r");
-      else
+      if (!canon)
 	canon = filename;
-    }
-  else
-    {
-      canon = filename;
-      file = fopen (filename, "r");
     }
 
+  if ((file_fd = open(canon, O_RDONLY | O_EXCL, 0022)) != -1)
+    file = fdopen (file_fd, "r");
+
   if (file == NULL)
     {
-      error (0, errno, _("Can't open configuration file %s"), canon);
+      if (opt_verbose)
+	error (0, errno, _("Can't open configuration file %s"), canon);
       if (canon != filename)
 	free ((char *) canon);
       return;
@@ -1082,12 +1080,14 @@
 
   if (!opt_only_cline)
     {
+      /* Always add the config paths first, since it may contain paths to
+       * override system libraries.  */
+      parse_conf (config_file);
+
       /* Always add the standard search paths.  */
       add_dir (SLIBDIR);
       if (strcmp (SLIBDIR, LIBDIR))
 	add_dir (LIBDIR);
-
-      parse_conf (config_file);
     }
 
   search_dirs ();
